### ISSUE TITLE
Question about  multi-source data joint trainig

### ISSUE THREAD

**Issue by wjn922** [zamling/PSALM#3]
Hi, I do have a question regarding the multi-dataset joint training.

During joint training, some gpus only have chat data, while other gpus have seg data. Then, in the same batch, the mask decoder would not be used in some gpus. The deepspeed would probably get hang in this situation, as in [issue](https://github.com/huggingface/accelerate/issues/2194).

How do you solve this problem? Thank you!

**Comment by zamling** [comment]
Hi,
I also met this question, so that I choose another way to joint training: only do one task in a iteration. The detailed codes are [here](https://github.com/zamling/PSALM/blob/4a078fa48f8e5a8b0f5bb853331fd5e004c40703/psalm/train/train_datasets.py#L771C9-L771C29).

**Comment by wjn922** [comment]
Got it. Thanks for your quick response.
